File: ecommerce/src/main/java/com/bautista/model/Review.java
package com.bautista.model;

import lombok.Data;
import java.util.Date;

@Data
public class Review {
    private Integer id;
    private Integer productId;
    private Integer userId;
    private Integer orderId;
    private Integer rating;
    private String title;
    private String comment;
    private Boolean verifiedPurchase;
    private Boolean isApproved;
    private String customerName;
    private String customerEmail;
    private Integer helpfulCount;
    private Integer notHelpfulCount;
    private Date createdAt;
    private Date updatedAt;
}

File: product/src/app/model/review.ts
export interface Review {
  id?: number;
  productId: number;
  userId: number;
  orderId?: number;
  rating: number;
  title?: string;
  comment?: string;
  verifiedPurchase?: boolean;
  isApproved?: boolean;
  customerName: string;
  customerEmail: string;
  helpfulCount?: number;
  notHelpfulCount?: number;
  createdAt?: Date;
  updatedAt?: Date;
}

export interface ReviewRequest {
  productId: number;
  rating: number;
  title?: string;
  comment?: string;
}

export interface ReviewSummary {
  productId: number;
  averageRating: number;
  totalReviews: number;
  fiveStarCount: number;
  fourStarCount: number;
  threeStarCount: number;
  twoStarCount: number;
  oneStarCount: number;
}